python/tvm/autotvm/task/space.py

```diff
@@ -226,7 +226,13 @@ def __init__(self, axes, policy, **kwargs):
     def _generate_space(self, now, tmp_stack, enforce_no_tail=False):
         """Generate space by DFS"""
         if now == self.num_output - 1:
-            prod = np.prod(tmp_stack, dtype=np.int64)
+            prod = 1
```

Comment:
It seems to me that manually implementing a classic array production is not necessary in any case. Since limited types are only enforced in numpy and Python's types are unlimited, it would be more concise to use Python builtins to calculate the product:
```python
import functools
import operator
prod = functools.reduce(operator.mul, tmp_stack, 1)
```

Note that the length of `tmp_stack` is always small (currently 4 at most, and I don't think it would be longer than 10), so this won't hurt the performance.

Reply:
It looks more pythonic. The number of lines will be similar. Here is one debate on this: :) 

https://stackoverflow.com/questions/9474412/python-alternative-to-reduce

Reply:
Well, I know that some people prevent `reduce` and other patterns like `map` from being used in Python 3, although I personally think not like `map` and `filter`, `reduce` is hard to be replaced by other pythonic representation. I am fine with that if you do not want to introduce `reduce` to the code. In this case I would suggest making this logic a standalone function in `autotvm/util.py`.

Reply:
Utility function looks like a overkill. A reduce call is fine. Updated. Thanks!